### ISSUE TITLE
Use natural ordering for files when playing files from a folder.

### ISSUE THREAD

**Issue by luckydonald** [iina/iina#3342]
### What you want IINA to do:
When playing files in a folder structure like this

> - `the_opened_folder/`
>    - `1. Introduction/`
>      - `FOO-01_1_Introduction.mp4`
>    - `2. foobar/`
>       - `FOO-01_2_I_am_the_second_one_not_number_twelve.mp4`
>    - _..._
>    - `9. batsbatz/`
>       -   `FOO-01_9_Almost_ten.mp4`
>     - `10. Wowsa/`
>       - `FOO-01_10_Now_this_is_where_the_bug_happens.mp4`
>     - `11. Another one`
>         - `FOO-10_Untitled.mp4`
Now I open `the_opened_folder/` in IINA.

It should play the contents in the order of **natural ordering:** 
 `1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23`.

### What IINA does currently:
It plays it in **alphanumerical order** ,
 `1 10 11 12 13 14 15 16 17 18 19 2 20 21 22 23 3 4 5 6 7 8 9`...

### Why you think this should be added:
Videos should be in the correct order.

**Examples of other projects that have something similar:**

**Comment by dmattera** [comment]
I could be wrong but I believe this is an issue with how macOS reports back the contents of folder.

example:

I have a folder `untitled folder` with 6 files:

- 1.png
- 2.png
- 3.png
- 10.png
- 20.png
- 30.png


running the `ls` command in Terminal to list the contents of this folder yields the files listed in alphanumerical order by default:

<img width="288" alt="Screen Shot 2021-03-09 at 10 02 39 AM" src="https://user-images.githubusercontent.com/13613522/110491138-f9eafa80-80be-11eb-813d-41054ed0240c.png">

In essence, my guess is that IINA/mpv currently just uses whatever ordering of files the OS reports back to it (as opposed to using it's own logic that purposefully orders the files this way) and so it would need to implement custom logic in order to achieve this.

**Comment by anohren** [comment]
> it would need to implement custom logic in order to achieve this.

No, I believe localized collation a la Finder is readily available, if there's a will.
